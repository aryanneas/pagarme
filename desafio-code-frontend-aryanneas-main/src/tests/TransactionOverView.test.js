import React from 'react'
import { render } from '@testing-library/react'

import Transactionsontext from '../pages/TransactionContext'
import TransactionOverview from '../components/TransactionOverView'

const TRANSACTIONS = [
  {
    amount: 10000,
    buyer_document: '12345678912',
    credit_card_cvv: '123',
    credit_card_expiration_date: '0121',
    credit_card_holder_name: 'JOAO S SAURO',
    credit_card_number: '4111111111111111',
    date: '2020-11-10T19:43:56.451Z',
    id: 1,
    status: 'paid',
  },
]

const ComponentProvider = () => (
  <Transactionsontext.Provider
    value={{
      transactions: TRANSACTIONS,
      setTransactions: () => {},
    }}
  >
    <TransactionOverview />
  </Transactionsontext.Provider>
)

describe('TransactionOverView page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ComponentProvider />)
    expect(asFragment()).toMatchSnapshot()
  })
})
