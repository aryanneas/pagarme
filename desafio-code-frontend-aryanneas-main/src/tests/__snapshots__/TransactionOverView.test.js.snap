// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionOverView page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fZdKfR"
  >
    <p
      class="sc-gsTCUz jcdWbx"
    >
      Número de transações
    </p>
    <p
      class="sc-dlfnbm irvmMI"
    >
      1
    </p>
    <p
      class="sc-gsTCUz jcdWbx"
    >
      Valor total
    </p>
    <p
      class="sc-dlfnbm irvmMI"
    >
      R$ 10.000,00
    </p>
  </div>
</DocumentFragment>
`;
